### DISCUSSION TITLE
Issue with a Repeater - Property type not supported in Livewire for property: ["01J49HGYDMRPK5YR4YBDEKZWFX"]

### DISCUSSION THREAD

**Discussion by acoustep** [filamentphp/filament#13792] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.97

### How can we help you?

This has worked previously, so I'm not sure what has changed that's causing this problem.

I have this error when I try to save my form:

`Property type not supported in Livewire for property: ["01J49HGYDMRPK5YR4YBDEKZWFX"]` , 

It only happens when I add a new row in the repeater. If I edit the rest of my form, it saves fine. If I remove a row in the repeater, it also saves fine. Livewire does not seem to like that I've added a new row.

This ID (`01J49HGYDMRPK5YR4YBDEKZWFX`) matches the ID of the newly created related invitation.

Here's some context, in my EventResource, I have a repeater for invitations

```php
//...
    public static function form(Form $form): Form {
        return $form
            ->columns(1)
            ->schema([
                Tabs::make()->tabs([
                   Tabs\Tab::make('Invitations')->schema([
                        Repeater::make('contacts')
                                ->reorderable(false)
                                ->columns(2)
                                ->collapsible(false)
                                ->collapsed(false)
                                ->defaultItems(0)
                                ->hiddenLabel()
                                ->cloneable()
                                ->deletable()
                                ->deleteAction(
                                    fn(Action $action) => $action->requiresConfirmation(),
                                )
                                ->addActionLabel('Add invitation')
                                ->relationship('invitations')
                                ->schema([
                                    Select::make('contact_id')
                                          ->label('Contact')
                                          ->searchable()
                                          ->preload()
                                          ->searchDebounce(500)
                                          ->relationship('contact', 'name',
                                              modifyQueryUsing: fn(Builder $query) => $query->whereTeamId(Filament::getTenant()->id)
                                          )
                                          ->required(),
                                          // More generic fields here
                                ])

//...
```

The new row is saved (hence the ULID), but it's bringing up the livewire error. Is there any way around this or does anyone know what I might be doing wrong?

Thanks

**Comment by leandrocfe** [comment]
Are you using the panel builder or only the form builder?

**Reply by acoustep** [reply]
Yes, I am using the panel builder.

**Reply by leandrocfe** [reply]
latest filament version?

**Reply by acoustep** [reply]
I'm on 3.2.97, which I believe is the latest.

**Comment by acoustep** [comment, marked as answer]
Reverting back to 3.2.92 has resolved this for me. I assume there's a bug introduced between 3.2.92 and 3.2.97, I haven't had a chance to look into it further.

**Comment by ismaail** [comment]
Downgrading to 3.2.95 works as well.

The issue starts in 3.2.96

The source of the issue is the commit 18da9f47f9cfe0e4afb5b1ac01124a078c90ee1c in the line.

---

For me the problem is with the Model attribute `ulid` which I fill using the `creating` method in the Model Observer.

```php
$model->ulid = Str::ulid();
```

---

A solution, in the Model cast the `ulid` attribute to `string`.

```php
protected function casts(): array
{
    return [
        'ulid' => 'string',
    ];
}
```

What do guys think, is this a good solution ?
